TechtonicaModLoader/MyClasses/Managers/ModManager.cs

```diff
@@ -134,7 +139,14 @@ public static void Load() {
                         mod.configFileLocation = mod.configFileLocation.Replace(ProgramData.FilePaths.bepInExConfigFolder, ProgramData.FilePaths.configsFolder);
                     }
 
-                    AddMod(mod);
+                    // Only add mods that are newer than the 1.0 release date or on the allowed list.
+                    if (mod.dateUpdated >= techtonicaReleaseDate || allowedMods.Contains(mod.name)) {
+                        AddMod(mod);
+                    }
+                    else {
+                        File.Delete(mod.zipFileLocation);
+                        DeleteMod(mod);
+                    }
```

Comment:
Few more steps involved with removing a broken mod:
- Uninstalling the mod
- Delete the mod from the current profile

Difficulty is that profiles haven't loaded by this point, so it would be better to add a function that gets called at the end of MainWindow.OnProgramLoaded() that loops through mods in the current profile and uninstalls them if they're broken ones.

Don't forget to then notify the user with `GuiUtils.ShowWarningMessage()` with a list of removed mods

Reply:
OK, I see what needs to happen.  Seems it's like InstalledModPanel.OnDeleteModClicked()?

Just need to figure out a natural place for this code.

Shouldn't we look through all profiles and not the active one?